src/libraries/System.Private.Uri/src/System/UriBuilder.cs

```diff
@@ -255,6 +255,26 @@ public Uri Uri
 
         public override int GetHashCode() => Uri.GetHashCode();
 
+        private string EncodeUserName(string input)
```

Comment:
```suggestion
        private static string EncodeUserInfo(string input)
```

Reply:
Thanks for the quick review! I should have time later today to work on getting these changes implemented and will update my PR once complete.

Reply:
I've updated `UriBuilder` with the recommended changes and have added in a couple of unit tests for `UriBuilderTests`, I am having trouble getting dotnet set up locally for running the build (issues with clang on my end I believe) but wanted to push this up for review, let me know if anything looks like it needs to be changed!